src/command/import.rs

```diff
@@ -56,10 +64,75 @@ impl Cmd {
 
             Self::Zsh => import_zsh(db).await,
             Self::Bash => import_bash(db).await,
+            Self::Resh => import_resh(db).await,
         }
     }
 }
 
+async fn import_resh(db: &mut (impl Database + Send + Sync)) -> Result<()> {
+    let histpath = std::path::Path::new(std::env::var("HOME")?.as_str()).join(".resh_history.json");
+
+    println!("Parsing .resh_history.json...");
+    #[allow(clippy::filter_map)]
+    let history = std::fs::read_to_string(histpath)?
+        .split('\n')
+        .map(str::trim)
+        .map(|x| serde_json::from_str::<ReshEntry>(x))
+        .filter_map(Result::ok)
+        .map(|x| {
+            #[allow(clippy::cast_possible_truncation)]
+            #[allow(clippy::cast_sign_loss)]
+            let timestamp = {
+                let secs = x.realtime_before.floor() as i64;
+                let nanosecs = (x.realtime_before.fract() * 1_000_000_000_f64).round() as u32;
```

Comment:
One small thing to confirm here. Is there any chance that Resh will have many history items with the same timestamp?

Due to how sync works at the moment, if too many items have the same timestamp, they will be missed while syncing. 

This is why the ZSH importer applies a small millisecond offset while importing data: https://github.com/ellie/atuin/blob/main/atuin-client/src/import/zsh.rs#L116

In the future I'd like to remove this by either 

1) Move the responsibility for this offset away from the importer
2) Improve sync so that this isn't an issue

Reply:
After having a look through my 150MiB resh_history.json, it appears that no two entries share a realtimeBefore, so this is not an issue.

Reply:
Fantastic! thanks